### PR TITLE
Kaitie new section advanced settings

### PR DESCRIPTION
**Pull request by kobryan0619**
This creates the "Advanced Settings" portion of the Section Setup page. 

image

## Links

- Jira ticket [link](https://codedotorg.atlassian.net/jira/software/projects/TEACH/boards/65?selectedIssue=TEACH-113)

## Testing story

This is just the start of the tests for this component as it is not currently linked up to actually modify sections yet.   BUT I am new to testing, so let me know if you think further needed at this point.

## Follow-up work

- Once the curriculum quick assign is ready to go, we can link up the conditional logic for showing the right Advanced Settings better.

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  updateSection,
+  section,
+
+  // these are currently hard coded into the parent component
+  // In the future we will want to make these dynamic based on the course
+  assignedUnitTextToSpeechEnabled,
+  assignedUnitLessonExtrasAvailable
+}) {
+  const handlePairProgrammingToggle = e => {
+    e.preventDefault;
```
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx]: if we need this, it should have parenthesis, like `e.preventDefault()`. but if it's working without the parens, maybe this line can be taken out entirely?

File: apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx [hunk 2]
```diff
@@ -33,6 +33,7 @@ export default function SingleSectionSetUp({
         values={section.grades || []}
         setValues={g => updateSection('grades', g)}
       />
+      <hr />
```
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx]: was this a merge issue, or did you put it back intentionally? i remember taking this out when we decided to not do multiple sections at once, so i just wanted to check

Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx]: Ohh... I did NOT put it back intentionally... so I am guessing it is a merge issue...

Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx]: When working on this, I had a couple of issues with merging staging with my branch, but honestly, this one doesn't look familiar to me - I don't remember resolving it by hand, which makes me a bit nervous.  I might need to back up and check to see if anything else got overwritten...

File: apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
+import ToggleSwitch from '@cdo/apps/code-studio/components/ToggleSwitch';
+// import {
+//   assignedUnitTextToSpeechEnabled,
+//   assignedUnitLessonExtrasAvailable
+// } from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
```
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx]: are these comments left in intentionally?

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx [hunk 4]
```diff
@@ -92,6 +112,42 @@ export default function SectionsSetUpContainer() {
...
         sectionCourse={sections[0].course}
       />
+      {/* <Button
+        id={'uitest-advanced-settings'}
+        className={moduleStyles.buttonStyle}
+        text={i18n.advancedSettings()}
+        size={Button.ButtonSize.large}
+        icon={caret}
+        onClick={toggleAdvancedSettingsOpen}
+      /> */}
```
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: are these comments left in intentionally?

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx [hunk 5]
```diff
@@ -72,8 +82,18 @@ const saveSection = (e, section) => {
     });
 };
 
+// add prop to indicate where this is coming from
```
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: is this a `TODO` comment?

**Review by rshipp — APPROVED** [review]
approving with some questions about commented things. the tests look fine to me, without having set up the save-button stuff yet.

File: apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx (hunk 2 above)
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx]: I ended up removing these by hand.

File: apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx (hunk 3 above)
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx]: No... I forgot to do a pass for comments.  I went back to remove these and clean up other comments.

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx (hunk 4 above)
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: Removed!

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx (hunk 5 above)
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: Yes, this was part of my clean up efforts too.

**Review by rshipp — APPROVED** [review]

**Pull request merged**